contracts/bridge/src/fee-market/FeeMarket.sol

```diff
@@ -385,70 +377,131 @@ contract FeeMarket is IFeeMarket {
         _reward(VAULT, total_vault_reward);
     }
 
-    function _reward_and_unlock_ontime(
+    function _settle_order(
         uint256 key,
-        uint256 diff_time,
         address delivery_relayer,
         address confirm_relayer
-    ) private returns (uint256 delivery_reward, uint256 confirm_reward, uint256 vault_reward) {
-        // get the message fee from the last top N relayers
+    ) private returns (
+        uint256 delivery_reward,
+        uint256 confirm_reward,
+        uint256 vault_reward
+    ) {
+        Order memory order = orderOf[key];
+        require(orderOf[key].assignedTime > 0, "!exist");
+        // Get the message fee from the top N relayers
         uint256 message_fee = getOrderFee(key);
+        // Get slot index and slot price
+        (uint256 slot, uint256 slot_price) = _get_slot_price(key, message_fee);
+        // Message surplus = Message Fee - Slot price
+        uint256 message_surplus = message_fee - slot_price;
```

Comment:
Do we need to check overflow here?

Reply:
https://github.com/darwinia-network/darwinia-messages-sol/blob/a6ae7140e80d82598ef5e801e6a3946687607e9b/contracts/bridge/src/fee-market/FeeMarket.sol#L248

message_fee is the last top N assigned relayer, and also the max fee of the three assigned relayers.